src/exo/LoopIR_scheduling.py

```diff
@@ -2564,11 +2564,147 @@ def map_e(self, e):
         return e
 
 
+
+class _DoNormalize(LoopIR_Rewrite):
+    def __init__(self, proc):
+        self.C = Sym("temporary_constant_symbol")
+        super().__init__(proc)
+
+        self.proc = InferEffects(self.proc).result()
+
+    def concat_map(self, op, lhs, rhs):
+        if op == '+':
+            # if has same key: add value
+            common = { key: (lhs[key]+rhs[key]) for key in lhs if key in rhs }
+            return lhs | rhs | common
```

Comment:
Are the semantics of union such that the right-hand side of the union is guaranteed to overwrite the left in the event of a key clash?

Reply:
Yes!